### ISSUE TITLE
Can't get this to work properly in Xfce Terminal Emulator on Xubuntu 12.04.5 LTS

### ISSUE THREAD

**Issue by AdmiralCurtiss** [lyze/posh-git-sh#10]
Hi there!

My predicament is the following: I'm a Windows user who's recently been forced to use Linux for work. One of the biggest things I was missing was the posh-git Powershell scripts included in the Github desktop application, so I was happy when I found this Linux port. However, I can't quite get it to work.

First, I tried following the instructions in the readme. However, setting PROMPT_COMMAND in .bashrc like instructed gave me the following error message:

    bash: printf: missing unicode digit for \u

Which would then repeat for every command prompt line.

I then found the alternate instructions to replace PS1 in the git-prompt.sh script itself. I did so, adding the following line near the end of .bashrc:

    PS1=$PS1'$(__git_ps1 " [(%s)]")\$ '

Which worked, sort of. It always just printed the current branch and nothing else. I eventually figured out I was dealing with the same problem as #9, so I worked around that by renaming the __git_ps1 function to __git_ps1_2, which finally gave me the actual output, except with broken color commands:

    user@machine:~/path/to/repo [( [(\[\]\[\e[1;33m\] [\[\]\[\e[1;36m\]master\[\]\[\e[1;32m\] +1 ~0 -0\[\]\[\e[1;33m\] |\[\]\[\e[0;31m\] +0 ~4 -0\[\]\[\e[1;33m\]]\[\]\[\e[0;34m\]$\[\]\[\e[m\])])]$ 

Which is obviously not ideal. What am I doing wrong?

I'd appreciate any help very much.

**Comment by ardentsword** [comment]
i think this is a problem I ran into at some point as well. It has to do with single and double quotation marks if I'm not mistaken. You can see an example of how I got it working in my dotfiles repo: https://github.com/ardentsword/dotfiles/blob/master/bash/.bashrc line 93

**Comment by AdmiralCurtiss** [comment]
Thanks a lot! Sure enough, setting the PROMPT_COMMAND like this works:

    PS1FRONT='${debian_chroot:+($debian_chroot)}\[\033[01;32m\]\u@\h\[\033[00m\]:\[\033[01;34m\]\w\[\033[00m\]'
    PS1BACK='\$ '
    PROMPT_COMMAND='__git_ps1_2 "'$PS1FRONT'" "'$PS1BACK'";'

So what's the actual issue here? Do different shells interpret single and double quotes differently?

**Comment by ardentsword** [comment]
Yes, there is indeed a pretty big difference in how bash interprets the quotes. Especially concerning variables. I found a nice article here explaining this: http://www.howtogeek.com/howto/29980/whats-the-difference-between-single-and-double-quotes-in-the-bash-shell/

**Comment by lyze** [comment]
@AdmiralCurtiss Thanks for bringing up #9 again. It seems the shadowing is happening a lot, so I addressed #9 with a namespace prefix to all functions. I'm glad that you figured out a workaround and that you got your prompt to work too!
